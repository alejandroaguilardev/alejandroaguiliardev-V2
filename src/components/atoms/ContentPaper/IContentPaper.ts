import { ReactElement } from "react";

export interface IContentPaper {
	children: ReactElement | ReactElement[];
}
