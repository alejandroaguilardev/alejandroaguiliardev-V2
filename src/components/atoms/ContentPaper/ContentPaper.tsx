import { Grid, Container } from "@mui/material"
import { useThemeContext } from "../../../context";
import { IContentPaper } from "./IContentPaper"

export const ContentPaper = ({ children }: IContentPaper) => {
  const { theme } = useThemeContext();
  return (
    <Container maxWidth="lg">
      <Grid sx={{
        backgroundColor: 'background.paper',
        boxShadow: theme.mode === 'dark' ? '1px 1px 30px #263240' : '1px 1px 30px #ddd',
      }}
        px={10} py={10}
      >
        {children}
      </Grid>
    </Container>
  )
}
