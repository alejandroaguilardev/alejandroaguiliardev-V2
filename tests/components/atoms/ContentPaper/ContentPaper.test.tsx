import { render, screen } from "@testing-library/react";
import { ContentPaper } from "../../../../src/components/atoms/ContentPaper/ContentPaper";

describe("ContentPaper Test", () => {
	test("Snapshots", () => {
		const { container } = render(
			<ContentPaper>
				<h1>ContentPaper</h1>
			</ContentPaper>
		);

		expect(container).toMatchSnapshot();
		expect(screen.getByText("ContentPaper")).toBeTruthy();
	});
});
