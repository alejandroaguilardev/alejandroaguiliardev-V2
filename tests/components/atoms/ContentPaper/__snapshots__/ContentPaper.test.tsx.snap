// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentPaper Test Snapshots 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <div
      class="MuiGrid-root css-hwjy3t-MuiGrid-root"
    >
      <h1>
        ContentPaper
      </h1>
    </div>
  </div>
</div>
`;
